src/main/haskell/kore/src/Kore/Predicate/Predicate.hs

```diff
@@ -79,6 +80,13 @@ can be manipulated only by functions in this module.
 newtype GenericPredicate pat = GenericPredicate pat
     deriving (Eq, Foldable, Functor, Generic, NFData, Ord, Show, Traversable)
 
+instance
+    (Hashable pat
+    ) => Hashable (GenericPredicate pat)
+  where
+    hashWithSalt s (GenericPredicate pat) =
```

Comment:
`Hashable` should provide a default implementation for `GenericPredicate` because it has a `Generic` instance.